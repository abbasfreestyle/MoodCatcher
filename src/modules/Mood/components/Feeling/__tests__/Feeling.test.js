import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Feeling } from '..';

const requiredProps = {
  feeling: { id: 1, name: 'Relaxed' },
  selected: false,
  onAddFeeling: jest.fn(),
  onRemoveFeeling: jest.fn()
};

const wrapper = shallow(<Feeling {...requiredProps} />);

describe('Modules > Mood > Feeling.', () => {
  describe('Snapshots.', () => {
    afterEach(() => {
      wrapper.setProps(requiredProps);
    });

    test('should match with required props', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should match when selected is true', () => {
      wrapper.setProps({ selected: true });
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
