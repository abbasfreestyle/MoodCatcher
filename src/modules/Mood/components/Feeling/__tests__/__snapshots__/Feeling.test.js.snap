// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modules > Mood > Feeling. Snapshots. should match when selected is true 1`] = `
<Feeling
  flex={true}
  margin={10}
  onPress={[Function]}
  selected={true}
>
  Relaxed
</Feeling>
`;

exports[`Modules > Mood > Feeling. Snapshots. should match with required props 1`] = `
<Feeling
  flex={true}
  margin={10}
  onPress={[Function]}
  selected={false}
>
  Relaxed
</Feeling>
`;
